src/main/java/com/epam/xmlparsing/entity/Flower.java

```diff
@@ -0,0 +1,70 @@
+package com.epam.xmlparsing.entity;
+
+import javax.xml.bind.annotation.*;
+
+@XmlRootElement(name = "flower", namespace = "plants")
+@XmlType(name = "flower", namespace = "plants")
+@XmlAccessorType(XmlAccessType.FIELD)
+public class Flower extends Plant {
+
+    @XmlElement(namespace = "plants")
+    private int petalNumber;
+
+    public Flower() {
+    }
+
+    public Flower(int id, String origin, String name, Soil soilType, int neededTemperature, boolean needsLight, int weeklyRequiredWater, int petalNumber) {
+        super(id, origin, name, soilType, neededTemperature, needsLight, weeklyRequiredWater);
+        this.petalNumber = petalNumber;
+    }
+
+    public Flower(FlowerBuilder builder) {
+        super(builder);
```

Comment:
builder pattern should be implemented other way, a good article regarding builder:
https://habr.com/ru/post/244521/